workflow/util/util.go

```diff
@@ -960,6 +974,11 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 				log.Debugf("Reset %s node %s since it's a group node", node.Name, string(node.Phase))
 				continue
 			} else {
+				if isDescendantNodeSuccessfully(wf, node) {
+					log.Debugf("Node %s remains as is since it has succeed nodes.", node.Name)
```

Comment:
`log.Debugf("Node %s remains as is since it has succeed child nodes.", node.Name)`